org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -539,4 +599,16 @@ public GradleCompatibilityInfo(String projectPath, String message, String highes
 			this.recommendedGradleVersion = recommendedGradleVersion;
 		}
 	}
+
+	private class UpgradeGradleWrapperInfo implements Serializable {
```

Comment:
what's purpose to implements `Serializable`?

Reply:
The `UpgradeGradleWrapperInfo` objects will be passed to the client side. But it seems other classes like `AccessorField` didn't implement `Serializable` and they work well. Seems redundant under LSP?

Reply:
lsp4j is using Gson library to serialize a class to json, no need of `Serializable` interface.